billing/views.py

```diff
@@ -130,3 +136,13 @@ def get_object(self):
                 .get(owner=self.request.user)
         except Account.DoesNotExist:
             raise Http404('No Account matches the given query.')
+
+
+@permission_classes([permissions.IsAuthenticated])
+@api_view(['POST'])
+def pay_debt(request):
+    account = request.user.billing_account
+    assign_funds_to_account_pending_invoices(account_id=account.id)
+    charge_pending_invoices(account_id=account.id)
+    status = HTTP_400_BAD_REQUEST if account.delinquent else HTTP_200_OK
+    return Response({'delinquent': account.delinquent}, status=status)
```

Comment:
Again, I think 'delinquent' is semantically strange: the user just tried to pay open invoices and we return something that is only very vaguely related.

Reply:
Updated